### ISSUE TITLE
Ensure tests are built with same dependencies as parent crate

### ISSUE THREAD

**Issue by dtolnay** [dtolnay/trybuild#53]
The crate that gets compiled by trybuild is given the same dependencies in its Cargo.toml as the outer crate has. But it may have an old Cargo.lock left over from a previous `cargo test` run that causes it to use a different set of patch versions of those dependencies compared to what is used currently by the outer crate whose tests we are running. To avoid this, trybuild should somehow write a Cargo.lock (or failing that, use a sequence of `cargo update --precise` commands) to keep the resolved set of dependencies in sync.

**Comment by TriplEight** [comment]
locally I'm getting
```
running 1 test
error: failed to download `aho-corasick v0.7.9`

Caused by:
  can't make HTTP request in the offline mode
test compile_tests ... FAILED
```

**Comment by ascjones** [comment]
A naive solution would be to remove the cached `Cargo.lock` from the location of the crate generated by `trybuild`.  e.g. https://github.com/dtolnay/trybuild/compare/master...ascjones:rm-cargo-lock.

It will then write a new `Cargo.lock` which will resolve the best set of dependencies available locally (because of `--offline`) which should be the same as those downloaded for the outer crate.

I've tested this for the repro above and it works, and rerunning the tests with the package resolution is fast. 

I'm not sure though that this will guarantee the exact same set of dependencies, or what other unintended consequences there may be. But putting this suggestion forward as the "simplest thing that will possibly work".

Another idea I'm currently investigating is when copying the dependencies to the crate generated by `trybuild`, read the precise versions from the outer crate's `Cargo.lock` and write them to the generated `Cargo.toml`. In this way we would guarantee the same set of dependencies. 

@dtolnay let me know what you think of those ideas and I can prepare a PR, or go back to the drawing board.

**Comment by ascjones** [comment]
As an alternative to deleting the `Cargo.lock`, running `cargo generate-lockfile --offline` would force refresh the Cargo.lock. 

Tried it and it works: https://github.com/dtolnay/trybuild/compare/master...ascjones:generate-lockfile

**Comment by dtolnay** [comment]
I published the partial workaround in #54 in 1.0.24. Thanks @ascjones!

**Comment by cecton** [comment]
I'm still getting this issue on version 1.0.35
